### ISSUE TITLE
Compilation fails on Fedora 21.

### ISSUE THREAD

**Issue by dpc** [rust-lang/rust#21215]
Since I upgraded to Fedora 21 from 20, rust compilation fails with:

```
cp: x86_64-unknown-linux-gnu/stage1/lib/librustc_resolve
cp: x86_64-unknown-linux-gnu/stage1/lib/librustc_borrowck
cp: x86_64-unknown-linux-gnu/stage1/lib/librustc_typeck
rustc: x86_64-unknown-linux-gnu/stage0/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_driver
cp: x86_64-unknown-linux-gnu/stage1/lib/librustc_trans
/home/dpc/opt/src/rust/src/librustc_driver/lib.rs:19:1: 19:15 warning: unused attribute, #[warn(unused_attributes)] on by default
/home/dpc/opt/src/rust/src/librustc_driver/lib.rs:19 #![staged_api]
                                                     ^~~~~~~~~~~~~~
rustc: x86_64-unknown-linux-gnu/stage0/lib/rustlib/x86_64-unknown-linux-gnu/bin/rustc
cp: x86_64-unknown-linux-gnu/stage1/lib/librustc_driver
cp: x86_64-unknown-linux-gnu/stage1/bin/rustc
rustc: x86_64-unknown-linux-gnu/stage1/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore
x86_64-unknown-linux-gnu/stage1/bin/rustc: symbol lookup error: x86_64-unknown-linux-gnu/stage1/bin/rustc: undefined symbol: _ZN2rt10lang_start20h9ceff13e315e3c7di0zE
/home/dpc/opt/src/rust/mk/target.mk:165: recipe for target 'x86_64-unknown-linux-gnu/stage1/lib/rustlib/x86_64-unknown-linux-gnu/lib/stamp.core' failed
make: *** [x86_64-unknown-linux-gnu/stage1/lib/rustlib/x86_64-unknown-linux-gnu/lib/stamp.core] Error 127
```

**Comment by emanueLczirai** [comment]
tl;dr: make help is lying to me saying that I can run rust from the build directory(without installing it on the system), well kinda, because this works(for example): 
```
$ LD_LIBRARY_PATH="../lib/" ./rustc
$ pwd
/home/emacs/build/rust/x86_64-unknown-linux-gnu/stage2/bin
```
--------

I'm getting the "same" thing if I run this:

```bash
$ pwd
/home/emacs/build/rust/x86_64-unknown-linux-gnu/stage2/bin
$ ./compiletest
./compiletest: symbol lookup error: ./compiletest: undefined symbol: _ZN11collections4hash3map36HashMap$LT$K$C$$u{20}V$C$$u{20}S$GT$6resize10_FILE_LINE20ha626c175c55fc0ddEmoE

$ ./rustc
./rustc: symbol lookup error: ./rustc: undefined symbol: _ZN2rt10lang_start20hc5e53532c7982aa4P7zE
```

 That symbol `_ZN2rt10lang_start20hc5e53532c7982aa4P7zE` is in `libstd-*.so` and I'm thinking I need to find a way to tell `rustc` to find/use the .so from `../lib/`
 Or make sure rust is installed (`make install`? haven't tried this yet)

`make help` however says, it should run from the build directory:

```bash
  * all - The default. Build a complete, bootstrapped compiler.
          `rustc` will be in `${target-triple}/stage2/bin/`. Run it
          directly from the build directory if you like. This also
          comes with docs in `doc/`.
```

The stage0 rustc works though (this one seems static, it's 54 megs):

```bash
$ pwd
/home/emacs/build/rust/x86_64-unknown-linux-gnu/stage0/bin
$ ./rustc
Usage: rustc [OPTIONS] INPUT
...
```

I already have an earlier rust version installed in the system. If I remove it and then try to run the stage2 rust(which isn't installed, obviously):

```bash
$ pwd
/home/emacs/build/rust/x86_64-unknown-linux-gnu/stage2/bin

$ ./rustc
./rustc: error while loading shared libraries: librustc_driver-4e7c5e5c.so: cannot open shared object file: No such file or directory

#or as the make help says: run it directly from the build directory (taken literally)
$ pwd
/home/emacs/build/rust
$ ./x86_64-unknown-linux-gnu/stage2/bin/rustc
./x86_64-unknown-linux-gnu/stage2/bin/rustc: error while loading shared libraries: librustc_driver-4e7c5e5c.so: cannot open shared object file: No such file or directory
```

That file is in `./rust/x86_64-unknown-linux-gnu/stage2/lib/`. But rust is expecting it to be somewhere else (not in current directory that's for sure, tested!) which is here: `/usr/lib/`  (if ./configure --prefix was /usr  I guess)

But this works(for example): 
```
$ LD_LIBRARY_PATH="../lib/" ./rustc
Usage: rustc [OPTIONS] INPUT

Options:
...
$ pwd
/home/emacs/build/rust/x86_64-unknown-linux-gnu/stage2/bin
```

**Comment by jacek99** [comment]
any updates on this? I just installed rust beta on a new Fedora 21 box and I got this

rustc: error while loading shared libraries: librustc_driver-4e7c5e5c.so: cannot open shared object file: No such file or directory


not able to use rust on this box

**Comment by dpc** [comment]
I've switched completely to `rustup` and haven't tried since then. But is it possible, that you still have old `rustc` installed through `rustup` while compiling `rustc` from source? If I understood some suggestion given above, that might be potential reason for this failure.

**Comment by thePalindrome** [comment]
According to #23193, LD_LIBRARY_PATH needs to be set to /usr/local/lib but only on Fedora.

Maybe a similar distro quirk prevents compilation?

**Comment by steveklabnik** [comment]
Has anyone seen this issue lately? It's been over a year without any update, I wonder if it is stale.

**Comment by dpc** [comment]
I think it's OK to consider stale. Rust compiler, Fedora, rustup etc. moved forward.

**Comment by jacek99** [comment]
I verified with a brand new install on Fedora 23, seems to be fine now.
